core/services/gateway/handlers/functions/subscriptions.go

```diff
@@ -130,19 +130,15 @@ func (s *onchainSubscriptions) queryLoop() {
 
 		blockNumber := big.NewInt(0).Sub(latestBlockHeight, s.blockConfirmations)
 
-		updateLastKnownCount := func() {
+		if lastKnownCount == 0 || start > lastKnownCount {
 			count, err := s.getSubscriptionsCount(ctx, blockNumber)
 			if err != nil {
 				s.lggr.Errorw("Error getting subscriptions count", "err", err)
 				return
```

Comment:
So now if updating the count fails, we exit the whole iteration. This should be fine but maybe better to still continue..? What do you think?

Reply:
Agreed, changed